OrbitGgp/Client.cpp

```diff
@@ -76,13 +88,25 @@ outcome::result<QPointer<Client>> Client::Create(QObject* parent) {
   QProcess ggp_process{};
   ggp_process.setProgram("ggp");
   ggp_process.setArguments({"version"});
+
+  std::optional<Error> error;
+  QObject::connect(&ggp_process, &QProcess::errorOccurred, parent, [&ggp_process, &error] {
+    ERROR("Ggp command line process failed with error: %s",
+          ggp_process.errorString().toStdString());
+    error = Error::kCouldNotUseGgpCli;
+  });
```

Comment:
What situation does this slot cover that is not covered by the `if (ggp_process.exitStatus()...` below?
Since this is all synchronous I don't really understand when this could potentially be called while the if from below is NOT true.

Reply:
It took me some testing to realize that on if the process can't be started at all (e.g because ggp is not installed/ found in path), then the `if` branch is not taken. This happens only on windows as far as I can tell. 

I tested this by just renaming the executable name `ggp` to something that does not exit.

Reply:
Ah great. Thanks for the explanation. That makes sense.

I would say this needs at least a comment for explanation. How do you feel about adding an additional condition to the if instead?

This might work:

```
if (ggp_process.exitStatus() != QProcess::NormalExit || ggp_process.exitCode() != 0 || ggp_process.error() != QProcess::UnknownError) {
```

Reply:
This is even better, thank you!